### PR TITLE
fix: popover position bugs on scroll #968

### PR DESCRIPTION
**Pull request by pedro-martins-brisa**
#968

This change is intended to fix popover positioning bugs. Therefore, the popover stylizations were corrected so that only its utils calculates the TOP and LEFT of what the element will be rendered.

A positioning and repositioning service ('ionPosition') was also created, which aims to validate and set a valid position for the component/directive. It is applied to both the class and the directive that wants to have this behavior

[Gravação de tela de 2024-01-09 13-50-16.webm](https://github.com/Brisanet/ion/assets/138057813/d244be5e-f92f-4c9d-b7fe-c7313765c714)

### CODE REVIEW

**Comment by allan-chagas-brisa** [comment]
https://github.com/Brisanet/ion/assets/138060158/564daf65-28e0-4e1a-ada1-f254092bfe6e

I don't think the left calculations are considering the arrow size

File: projects/ion/src/lib/popover/popover.directive.ts [hunk 1]
```diff
@@ -154,21 +160,27 @@ export class IonPopoverDirective implements OnDestroy {
...
-    const position = hostElement.getBoundingClientRect();
-
-    if (this.elementIsEnabled(hostElement)) {
-      this.open(position);
+    if (this.elementIsEnabled(hostElement) && this.isComponentRefNull()) {
+      this.open();
     }
   }
 
+  @HostListener('window:scroll') onScroll(): void {
```
Comment (danilo-moreira-brisa) [inline projects/ion/src/lib/popover/popover.directive.ts:168]: To help to solve the problem where popover doesn't close correctly if is not window that has the scrollbar, we can add some more events, like `document:scroll`, `body:scroll` and, primarly, `window:wheel`, to listen or detect any scrolls on the page.

Comment (pedro-martins-brisa) [reply projects/ion/src/lib/popover/popover.directive.ts:168]: OK, I'll add these events

**Review by danilo-moreira-brisa — APPROVED** [review]
LGTM!

**Review by allan-chagas-brisa — APPROVED** [review]

**Pull request merged**